nylas/client/errors.py

```diff
@@ -17,3 +22,23 @@ class UnSyncedError(NylasError):
     """
 
     pass
+
+
+class NylasApiError(HTTPError):
+    """
+    Error class for Nylas API Errors
+    This class provides more information to the user sent from the server, if present
+    """
+
+    def __init__(self, response):
+        try:
+            response_json = json.loads(response.text)
+            error_message = u"%s %s. Reason: %s. Nylas Error Type: %s" % (
+                response.status_code,
+                response.reason,
+                response_json["message"],
+                response_json["type"],
+            )
+            super(NylasApiError, self).__init__(error_message, response=response)
+        except (ValueError, KeyError):
+            super(NylasApiError, self).__init__(response.text, response=response)
```

Comment:
this text could be large, and why would the json parse fail here?

Reply:
The reason why JSON parsing could fail is a rare occasion where the server is down, the error returned would be an HTML or just plain text, which would result in a paring error when attempting to parse. We've had this occur with the [Ruby SDK for example](https://github.com/nylas/nylas-ruby/pull/316).